Client/Frontend/Menu/AppMenuItem.swift

```diff
@@ -9,12 +9,25 @@ struct AppMenuItem: MenuItem {
     let action: MenuAction
     let secondaryAction: MenuAction?
     let animation: Animatable?
+    var isDisabled: Bool
     private let iconName: String
     private let privateModeIconName: String
     private let selectedIconName: String?
 
     private var icon: UIImage? {
-        return UIImage(named: iconName)
+        if isDisabled {
+            let imageView = UIImageView(image: UIImage(named: iconName))
+            imageView.image = imageView.image?.imageWithRenderingMode(.AlwaysTemplate)
+            imageView.tintColor = UIColor.lightGrayColor()
+            
+            UIGraphicsBeginImageContextWithOptions(imageView.bounds.size, imageView.opaque, 0.0)
+            imageView.layer.renderInContext(UIGraphicsGetCurrentContext()!)
+            let image = UIGraphicsGetImageFromCurrentImageContext()
+            UIGraphicsEndImageContext()
+            return image
+        } else {
+            return UIImage(named: iconName)
+        }
```

Comment:
Hm...looking more closely, I think it might actually make more sense to do this in `MenuViewController.swift`. A few reasons:
 * `AppMenuItem.swift` is app-specific, whereas this property is defined in `MenuItem.swift` which is menu-specific. That is, we've discussed eventually moving the generic menu code to a separate library/framework, so it would be nice if the app didn't have to manually handle drawing the disabled state.
 * We also want to grey the text, which we can't do here anyway.

Doing this in `MenuViewController.swift`, I think we can tint the `UIImageView` itself instead of creating a new graphics context. So, basically, something like this:
```
    func menuView(menuView: MenuView, menuItemCellForIndexPath indexPath: NSIndexPath) -> MenuItemCollectionViewCell {
        if let icon = menuItem.iconForState(appState) {
            cell.menuImageView.image = icon.imageWithRenderingMode(.AlwaysTemplate)
            if menuItem.isDisabled {
                cell.menuImageView.tintColor = menuConfig.disabledTintColor()
            }
        }
    }
```

And we would also want to use `menuConfig.disabledTintColor()` for the cell's text label.

Sorry for misleading you before -- I hope this makes sense. Let me know if you have any questions!